test/unit/util/time.spec.js

```diff
@@ -77,24 +118,58 @@ const testOverMonths = (expectedDatesFn, options = {}) => {
   for (const m of months) {
     const daysOfMonth = m === '02' ? 28 : 30
     // No DST switch happened on the 31th of any month in 2018.
-    const recurringDates = computeRecurringDates('0 0 * * *', {
-      startDate: `2018-${m}-01T00:00:00.000Z`,
-      endDate: `2018-${m}-${daysOfMonth}T00:00:00.001Z`,
+    const startDate = `2018-${m}-01T00:00:00.000Z`
+    const endDate = `2018-${m}-${daysOfMonth}T00:00:00.001Z`
+
+    const recurringDates = computeRecurringDates(pattern, {
+      startDate,
+      endDate,
       ...options
       // timezone: 'UTC' // should be the default
     })
 
-    expectedDatesFn({ m, daysOfMonth, recurringDates })
+    expectedDatesFn({ m, daysOfMonth, recurringDates, startDate, endDate })
   }
 }
 
 test('computes recurring dates with UTC by default, ignoring any local DST shift', (t) => {
-  testOverMonths(({ m, daysOfMonth, recurringDates }) => {
-    const expectedDates = _.range(daysOfMonth).map(padDayOrMonth)
-      .map(d => `2018-${m}-${d}T00:00:00.000Z`) // no DST shift
+  const getExpectedDates = ({ m, daysOfMonth, endDate, frequency = 'day' }) => {
+    if (frequency === 'day') {
+      return _.range(daysOfMonth).map(padDayOrMonth)
+        .map(d => `2018-${m}-${d}T00:00:00.000Z`)
+    } else if (frequency === 'hour') {
+      const dates = _.range(daysOfMonth).map(padDayOrMonth)
+        .map(d => `2018-${m}-${d}`)
+
+      return dates.reduce((expected, d) => {
+        const time = _.range(0, 24).map(padTime)
+          .map(h => `${h}:00:00.000Z`)
 
+        const fullDates = time.map(t => `${d}T${t}`)
+        return expected.concat(fullDates).filter(d => d < endDate)
+      }, [])
+    }
+  }
+
+  testOverMonths(({ m, daysOfMonth, recurringDates }) => {
+    const expectedDates = getExpectedDates({ m, daysOfMonth })
     t.deepEqual(recurringDates, expectedDates)
   })
+
+  testOverMonths(({ m, daysOfMonth, recurringDates }) => {
+    const expectedDates = getExpectedDates({ m, daysOfMonth })
+    t.deepEqual(recurringDates, expectedDates)
+  }, 'day', { timezone: null })
+
+  testOverMonths(({ m, daysOfMonth, recurringDates, endDate }) => {
+    const expectedDates = getExpectedDates({ m, daysOfMonth, frequency: 'hour', endDate })
+    t.deepEqual(recurringDates, expectedDates)
+  }, 'hour')
+
+  testOverMonths(({ m, daysOfMonth, recurringDates, endDate }) => {
+    const expectedDates = getExpectedDates({ m, daysOfMonth, frequency: 'hour', endDate })
+    t.deepEqual(recurringDates, expectedDates)
+  }, 'hour')
```

Comment:
These lines seem duplicated. Maybe you forgot timezone.

Reply:
Good catch. The timezone is missing.